src/Control/Algebra.hs

```diff
@@ -214,10 +225,26 @@ toRWSTF :: Monoid w => w -> (a, s, w) -> RWSTF w s a
 toRWSTF w (a, s, w') = RWSTF (a, s, mappend w w')
 {-# INLINE toRWSTF #-}
 
+newtype Swap s a = Swap { getSwap :: (a, s) }
+  deriving (Functor)
```

Comment:
This is handy for giving us a `Functor` instance for the backwards pairs `transformers` is so fond of.